### PR TITLE
Introduce editorconf checker linting - excluding MaxLineLength

### PR DESCRIPTION
**Pull request by javydekoning**
*Issue #, if available:*
#491 

*Description of changes:*
This introduces editorconf checker and a series of syntax fixes. This does NOT yet enforce `MaxLineLength`. 

`MaxLineLength` will be introduced in a later PR and is tracked via #491 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/organization_policy.py [hunk 1]
```diff
@@ -193,21 +182,18 @@ def apply(self, organizations, parameter_store, config):  # pylint: disable=R091
...
-                            current_policy.get('Content')
-                        ) == self._compare_ordered_policy(proposed_policy)
-                    ):
+                    if self._compare_ordered_policy(
+                        current_policy.get("Content")
+                    ) == self._compare_ordered_policy(proposed_policy):
                         LOGGER.info(
-                            'Policy (%s) %s does not require updating. '
-                            'Path is: %s',
+                            "Policy (%s) %s does not require updating. " "Path is: %s",
```
Comment (sbkok) [inline src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/organization_policy.py:189]: Same here:

```suggestion
                            "Policy (%s) %s does not require updating. Path is: %s",
```

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/organization_policy.py [hunk 2]
```diff
@@ -172,16 +162,15 @@ def apply(self, organizations, parameter_store, config):  # pylint: disable=R091
                         )
             except ParameterNotFoundError:
                 LOGGER.debug(
-                    'Parameter %s was not found in Parameter Store, '
-                    'continuing.',
+                    "Parameter %s was not found in Parameter Store, " "continuing.",
```
Comment (sbkok) [inline src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/organization_policy.py:165]: This looks wrong. If they are on the same line, it should be one string like so:

```suggestion
                    "Parameter %s was not found in Parameter Store, continuing.",
```

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/cdk/generate_pipeline_inputs.py [hunk 3]
```diff
@@ -76,46 +76,69 @@ def fetch_required_ssm_params(regions):
...
     pipeline = Pipeline(p)
-    if auto_create_repositories == 'enabled':
-        code_account_id = p.get('default_providers', {}).get('source', {}).get('properties', {}).get('account_id', {})
-        has_custom_repo = p.get('default_providers', {}).get('source', {}).get('properties', {}).get('repository', {})
-        if auto_create_repositories and code_account_id and str(code_account_id).isdigit() and not has_custom_repo:
-            repo = Repo(code_account_id, p.get('name'), p.get('description'))
+    if auto_create_repositories == "enabled":
+        code_account_id = (
+            p.get("default_providers", {})
+            .get("source", {})
```
Comment (sbkok) [inline src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/cdk/generate_pipeline_inputs.py:103]: I wasn't aware you could do this, nice.

**Review by sbkok — CHANGES_REQUESTED** [review]
Thanks for contributing this, great improvement!

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/cdk/generate_pipeline_inputs.py (hunk 3 above)
Comment (javydekoning) [reply src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/cdk/generate_pipeline_inputs.py:103]: FYI: https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#call-chains

**Review by sbkok — APPROVED** [review]

**Review by StewartW — APPROVED** [review]

**Pull request merged**